pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -38,7 +46,108 @@ func (srv PrometheusSrv) RouteGetAlertStatuses(c *models.ReqContext) response.Re
 }
 
 func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Response {
-	recipient := c.Params(":Recipient")
-	srv.log.Info("RouteGetRuleStatuses: ", "Recipient", recipient)
-	return response.Error(http.StatusNotImplemented, "", nil)
+	ruleResponse := apimodels.RuleResponse{
+		DiscoveryBase: apimodels.DiscoveryBase{
+			Status: "success",
+		},
+		Data: apimodels.RuleDiscovery{},
+	}
+
+	ruleGroupQuery := ngmodels.ListOrgRuleGroupsQuery{
+		OrgID: c.SignedInUser.OrgId,
+	}
+	if err := srv.store.GetOrgRuleGroups(&ruleGroupQuery); err != nil {
+		ruleResponse.DiscoveryBase.Status = "error"
+		ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting rule groups: %s", err.Error())
+		ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+		return response.JSON(http.StatusInternalServerError, ruleResponse)
+	}
+
+	for _, groupId := range ruleGroupQuery.Result {
+		alertRuleQuery := ngmodels.ListRuleGroupAlertRulesQuery{OrgID: c.SignedInUser.OrgId, RuleGroup: groupId}
+		if err := srv.store.GetRuleGroupAlertRules(&alertRuleQuery); err != nil {
+			ruleResponse.DiscoveryBase.Status = "error"
+			ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting rules for group %s: %s", groupId, err.Error())
+			ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+			return response.JSON(http.StatusInternalServerError, ruleResponse)
+		}
+
+		newGroup := &apimodels.RuleGroup{
+			Name:           groupId,
+			File:           "", // TODO: set this once we are setting it somewhere
+			LastEvaluation: time.Time{},
+			EvaluationTime: 0, // TODO: see if we are able to pass this along with evaluation results
```

Comment:
For a specific rule we just [log it](https://github.com/grafana/grafana/blob/eb06f846a03e123a22964713ecc771cb1877fb28/pkg/services/ngalert/schedule/schedule.go#L79) but not persisting it.
So this value could be longest of these durations.